v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/PostgreSQLJdbcValueMappings.java

```diff
@@ -19,34 +19,216 @@
+      };
+
+  // Value might be a Double.NaN or a valid BigDecimal
+  private static final ResultSetValueMapper<Number> numericToAvro =
+      (value, schema) -> value.toString();
+
+  private static final ResultSetValueMapper<java.sql.Date> dateToAvro =
       (value, schema) -> (int) value.toLocalDate().toEpochDay();
 
-  // TODO(thiagotnunes): Add missing type mappings
+  private static final ResultSetValueMapper<java.sql.Timestamp> timestampToAvro =
+      (value, schema) -> toMicros(value.toInstant());
+
+  private static final ResultSetValueMapper<OffsetTime> timeToAvro =
+      (value, schema) -> toMicros(value);
+
+  private static final ResultSetValueMapper<OffsetTime> timetzToAvro =
+      (value, schema) ->
+          new GenericRecordBuilder(TimeTz.SCHEMA)
+              .set(
+                  TimeTz.TIME_FIELD_NAME,
+                  TimeUnit.NANOSECONDS.toMicros(value.toLocalTime().toNanoOfDay()))
+              .set(
+                  TimeTz.OFFSET_FIELD_NAME,
+                  TimeUnit.SECONDS.toMillis(value.getOffset().getTotalSeconds()))
+              .build();
+
+  private static final ResultSetValueMapper<OffsetDateTime> timestamptzToAvro =
+      (value, schema) ->
+          new GenericRecordBuilder(TimeStampTz.SCHEMA)
+              .set(TimeStampTz.TIMESTAMP_FIELD_NAME, toMicros(value.toInstant()))
+              .set(
+                  TimeStampTz.OFFSET_FIELD_NAME,
+                  TimeUnit.SECONDS.toMillis(value.getOffset().getTotalSeconds()))
+              .build();
+
   private static final ImmutableMap<String, JdbcValueMapper<?>> SCHEMA_MAPPINGS =
       ImmutableMap.<String, Pair<ResultSetValueExtractor<?>, ResultSetValueMapper<?>>>builder()
           .put("BIGINT", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("BIGSERIAL", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("BIT", Pair.of(bytesExtractor, valuePassThrough))
+          .put("BIT VARYING", Pair.of(bytesExtractor, valuePassThrough))
+          .put("BOOL", Pair.of(ResultSet::getBoolean, valuePassThrough))
+          .put("BOOLEAN", Pair.of(ResultSet::getBoolean, valuePassThrough))
+          .put("BOX", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("BYTEA", Pair.of(bytesExtractor, valuePassThrough))
+          .put("CHAR", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("CHARACTER", Pair.of(ResultSet::getString, valuePassThrough))
           .put("CHARACTER VARYING", Pair.of(ResultSet::getString, valuePassThrough))
-          .put("DATE", Pair.of(utcDateExtractor, sqlDateToAvro))
+          .put("CIDR", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("CIRCLE", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("CITEXT", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("DATE", Pair.of(dateExtractor, dateToAvro))
+          .put("DECIMAL", Pair.of(ResultSet::getObject, numericToAvro))
```

Comment:
For PG decimal, we don't have a mapping [here](https://cloud.google.com/datastream/docs/unified-types), would it make sense to default to Mysql mapping then?

MySQL mapping is [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/MysqlJdbcValueMappings.java#L202)

Reply:
DECIMAL is an alias for NUMERIC.

I cannot do the same mapping as MySQL as PostgreSQL allows for NaN, which is not representable in a BigDecimal.